pkg/ebpf/bpf_client.go

```diff
@@ -573,7 +590,80 @@ func (l *bpfClient) GetDeletePodIdentifierLockMap() *sync.Map {
 }
 
-func (l *bpfClient) AttacheBPFProbes(pod types.NamespacedName, podIdentifier string) error {
+func (l *bpfClient) GetNetworkPolicyMode() string {
+	return l.networkPolicyMode
+}
+
+// loadIPAMDataFromFile reads IPAM JSON file from specified path and caches pod to interface count mapping
+func (l *bpfClient) loadIPAMDataFromFile(filePath string) error {
+	data, err := os.ReadFile(filePath)
+	if err != nil {
+		return fmt.Errorf("failed to read ipam.json file: %v", err)
+	}
+
+	var checkpointData datastore.CheckpointData
+	if err := json.Unmarshal(data, &checkpointData); err != nil {
+		return fmt.Errorf("failed to parse ipam.json: %v", err)
+	}
+
+	// Store interface count per pod from IPAM data
+	for _, entry := range checkpointData.Allocations {
+		// Skip entries with missing pod information
+		if entry.Metadata.K8SPodName == "" || entry.Metadata.K8SPodNamespace == "" {
+			continue
+		}
+
+		podNamespacedName := utils.GetPodNamespacedName(entry.Metadata.K8SPodName, entry.Metadata.K8SPodNamespace)
+		if entry.Metadata.InterfacesCount > 0 {
+			l.podNameToInterfaceCount.Store(podNamespacedName, entry.Metadata.InterfacesCount)
+			log().Debugf("Cached interface count for pod %s: %d", podNamespacedName, entry.Metadata.InterfacesCount)
+		}
+	}
+
+	log().Infof("Loaded IPAM data from %d allocations", len(checkpointData.Allocations))
+	return nil
+}
+
+// getInterfaceCountFromBackupFile attempts to read interface count from ipam.json loaded in podNameToInterfaceCount cache
+func (l *bpfClient) getInterfaceCountFromBackupFile(pod types.NamespacedName, podIdentifier string) (int, error) {
+	podNamespacedName := utils.GetPodNamespacedName(pod.Name, pod.Namespace)
+	if count, ok := l.podNameToInterfaceCount.Load(podNamespacedName); ok {
+		return count.(int), nil
+	}
+	return 0, errors.New("interface count not found in podNameToInterfaceCount cache")
+}
+
+// getInterfaceCountForPod determines the number of interfaces for a pod
+// Returns the interface count or an error if it cannot be determined
+func (l *bpfClient) getInterfaceCountForPod(pod types.NamespacedName, podIdentifier string, providedCount int) (int, error) {
+	// If interface count is provided (from RPC call), use it
+	if providedCount > 0 {
+		return providedCount, nil
+	}
+
+	// If multi-NIC is not enabled, default to 1 interface
+	if !l.isMultiNICEnabled {
+		return INTERFACE_COUNT_DEFAULT, nil
+	}
+
+	// Multi-NIC is enabled but we don't have interface count
+	// Try to get interface count from podNameToInterfaceCount cache
+	backupInterfaceCount, err := l.getInterfaceCountFromBackupFile(pod, podIdentifier)
+	if err == nil && backupInterfaceCount > 0 {
+		log().Infof("Found interface count %d from cache for pod %s", backupInterfaceCount, pod.Name)
+		return backupInterfaceCount, nil
+	}
+
+	// No interface count available and multi-NIC enabled - skip attachment
+	return 0, errors.New("Skipping probe attach: multiNIC enabled and interface count is unknown")
+}
```

Comment:
should we move this out to something like `PodInterfacesProvider` and pass that to `bpfclient` constructor?

Thinking in terms of separating concerns and overloading `bpfclient` domain.